etc/convert-test-results.swift

```diff
@@ -0,0 +1,332 @@
+import Foundation
+
+/// Represents a test suite.
+struct TestSuite {
+    /// Name of the test suite.
+    let name: String
+
+    /// Total execution time for the test suite, in seconds.
+    let time: TimeInterval
+
+    /// Tests in the suite.
+    let tests: [TestCase]
+
+    /// Count of tests in the suite.
+    let count: Int
+
+    /// Count of failed tests in the suite.
+    let failureCount: Int
+
+    /// Converts this test suite to XML.
+    func toXML() -> String {
```

Comment:
I considered actually using an XML encoder as there are a few Codable-style ones floating around, but that would require making this a library rather than a standalone script. since the XML is so simple it didn't seem worth it.